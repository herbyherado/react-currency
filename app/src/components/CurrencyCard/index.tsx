import React, { PureComponent } from 'react';
import './index.css';

const localeOptions = {
  minimumFractionDigits: 2,
  maximumFractionDigits: 4
};

class CurrencyCard extends PureComponent<CurrencyCardProps> {
  handleRemoveCard = () => {
    this.props.onRemoveCard();
  };

  render() {
    const { currencyName, currencyValue, currencyFullName, baseName, baseValue } = this.props;

    return (
      <div className="card">
        <div className="card-content">
          <div className="card-name">
            <p>{currencyName}</p>
            {currencyValue ? (
              <p className="card-number">
                {(currencyValue * baseValue).toLocaleString('en', localeOptions)}
              </p>
            ) : null}
          </div>
          <div className="card-footer">
            <p className="text-italic text-bolder">
              {currencyName} - {currencyFullName}
            </p>
            <p className="text-italic">
              1 {baseName} = {currencyName}{' '}
              {currencyValue ? currencyValue.toLocaleString('en', localeOptions) : null}
            </p>
          </div>
        </div>
        <button className="card-delete" onClick={this.handleRemoveCard}>
          -
        </button>
      </div>
    );
  }
}

type CurrencyCardProps = {
  currencyName: string;
  currencyFullName: string;
  currencyValue: number;
  baseName: string;
  baseValue: number;
  onRemoveCard: () => void;
};

export default CurrencyCard;
