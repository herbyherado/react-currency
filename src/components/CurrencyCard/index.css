/* CURRENCY BOX */

div.card {
  border: 1px solid white;
  margin: 1rem 0;
  display: flex;
  flex-direction: row;

  overflow: hidden;
}

div.card-content {
  width: 100%;
  flex: 0 0 90%;
  padding: 0.5rem;

  white-space: nowrap;
  text-overflow: clip;
  overflow: hidden;
}

button.card-delete {
  flex: 0 0 10%;
  border: 1px solid white;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  background: transparent;
  color: #ffffff;
}

div.card-name {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  margin-bottom: 0.5rem;
}

div.card-name p {
  font-size: 1.25rem;
}

div.card-name p.card-number {
  width: 100%;
  text-align: right;
  min-width: 0;

  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

div.card-footer p {
  margin: 0.25rem 0;
  font-size: 0.9rem;
}
