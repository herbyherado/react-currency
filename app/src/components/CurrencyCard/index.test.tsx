import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyCard from './index';

describe('CurrencyCard component testing', function() {
  describe('CurrencyCard snapshot testing', function() {
    it('should match to snapshot', function() {
      const component = renderer.create(
        <CurrencyCard
          currencyName="CAD"
          currencyFullName="Canadian Dollar"
          currencyValue={10020}
          baseName="USD"
          baseValue={20}
          onRemoveCard={jest.fn()}
        />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('CurrencyCard method calls', function() {
    describe('handleRemoveCard', function() {
      it('should call onRemoveCard props', function() {
        const component = renderer.create(
          <CurrencyCard
            currencyName="CAD"
            currencyFullName="Canadian Dollar"
            currencyValue={10020}
            baseName="USD"
            baseValue={20}
            onRemoveCard={jest.fn()}
          />
        );

        const handleRemoveCard = component.root.instance.handleRemoveCard;
        const propsSpy = jest.spyOn(component.root.props, 'onRemoveCard');

        handleRemoveCard();
        expect(propsSpy).toHaveBeenCalled();
      });
    });
  });
});
