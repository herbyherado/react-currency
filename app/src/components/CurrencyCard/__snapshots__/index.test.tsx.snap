// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyCard component testing CurrencyCard snapshot testing should match to snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="card-name"
    >
      <p>
        CAD
      </p>
      <p
        className="card-number"
      >
        200,400.00
      </p>
    </div>
    <div
      className="card-footer"
    >
      <p
        className="text-italic text-bolder"
      >
        CAD
         - 
        Canadian Dollar
      </p>
      <p
        className="text-italic"
      >
        1 
        USD
         = 
        CAD
         
        10,020.00
      </p>
    </div>
  </div>
  <button
    className="card-delete"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
